internal/pkg/cli/pipeline_show.go

```diff
@@ -6,53 +6,166 @@ package cli
+		return nil, fmt.Errorf("connect to environment datastore: %w", err)
+	}
+
+	ws, err := workspace.New()
+	if err != nil {
+		return nil, fmt.Errorf("workspace cannot be created: %w", err)
+	}
+
+	p := session.NewProvider()
+	defaultSession, err := p.Default()
+	if err != nil {
+		return nil, err
+	}
+
+	opts := &showPipelineOpts{
+		showPipelineVars: vars,
+		ws:               ws,
+		store:            ssmStore,
+		pipelineSvc:      codepipeline.New(defaultSession),
+	}
+
+	return opts, nil
 }
 
 // Validate returns an error if the flag values passed by the user are invalid.
 func (o *showPipelineOpts) Validate() error {
-	// TODO Placeholder
+	if o.ProjectName() != "" {
+		if _, err := o.store.GetProject(o.ProjectName()); err != nil {
+			return err
+		}
+	}
+	if o.pipelineName != "" {
+		if _, err := o.pipelineSvc.GetPipeline(o.pipelineName); err != nil {
+			return err
+		}
+	}
+
 	return nil
 }
 
 // Ask prompts for fields that are required but not passed in.
 func (o *showPipelineOpts) Ask() error {
-	// TODO Placeholder
+	if err := o.askProject(); err != nil {
+		return err
+	}
+
+	return o.askPipelineName()
+}
+
+func (o *showPipelineOpts) askProject() error {
+	if o.ProjectName() != "" {
+		return nil
+	}
+	projNames, err := o.retrieveProjects()
+	if err != nil {
+		return err
+	}
+	if len(projNames) == 0 {
+		return fmt.Errorf("no project found: run %s please", color.HighlightCode("project init"))
+	}
+	proj, err := o.prompt.SelectOne(
+		pipelineShowProjectNamePrompt,
+		pipelineShowProjectNameHelpPrompt,
+		projNames,
+	)
+	if err != nil {
+		return fmt.Errorf("select projects: %w", err)
+	}
+	o.projectName = proj
+
 	return nil
 }
 
-// Execute writes the pipeline manifest file.
-func (o *showPipelineOpts) Execute() error {
-	// TODO Placeholder
+func (o *showPipelineOpts) retrieveProjects() ([]string, error) {
+	projs, err := o.store.ListProjects()
+	if err != nil {
+		return nil, fmt.Errorf("list projects: %w", err)
+	}
+	projNames := make([]string, len(projs))
+	for ind, proj := range projs {
+		projNames[ind] = proj.Name
+	}
+	return projNames, nil
+}
+
+func (o *showPipelineOpts) askPipelineName() error {
+	// return if pipeline name is set by flag
+	if o.pipelineName != "" {
+		return nil
+	}
+
+	pipelineName, err := o.getPipelineNameFromManifest()
+	fmt.Println(pipelineName)
+	if err != nil {
+		return err
+	}
+
+	o.pipelineName = pipelineName
+	// if pipelineName == "" {
+	// get pipeline name from remote
+	// }
+
 	return nil
 }
 
-func newShowPipelineOpts(vars showPipelineVars) (*showPipelineOpts, error) {
-	ws, err := workspace.New()
+func (o *showPipelineOpts) getPipelineNameFromManifest() (string, error) {
+	// try to read pipelinename from from manifest
+
+	data, err := o.ws.ReadPipelineManifest()
 	if err != nil {
-		return nil, fmt.Errorf("workspace cannot be created: %w", err)
+		log.Infof("No pipeline manifest in workspace for project %s\n.", color.HighlightUserInput(o.ProjectName()))
+		return "", nil
```

Comment:
We could return an error here and let the caller (`askPipelineName`) decide to ignore it and log the info. 
This way the caller doesn't need to check if the name returned is not "" when err is nil.

Reply:
Sure, this was kind of pending the changes I was making here: https://github.com/aws/amazon-ecs-cli-v2/commit/f4f65449e3a5a97e5945df16c796319f41deae46. I feel like for now this was the easiest thing to do until the custom error is returned. I can add a note to change it once the other branch is merged.